src/app/pages/existing-query/existing-query.component.ts

```diff
@@ -104,15 +104,18 @@ export class ExistingQueryComponent implements OnInit {
   onFinalizeSubmit(): void {
     if (confirm(`Are you sure you would like to send this query to be finalized?`)) {
       this.canvasLoading = true;
-      this.existingQueryService.updateQueryStateToProcessFinalized()
-        .then(() => {
-          const message = `"${this.existingQueryService.currentQuery.name}": has been submitted to be finalized`;
-          this.alertService.setAlert(message, AlertType.Success);
-          this.canvasLoading = false;
-        })
-        .catch(() => {
-          this.canvasLoading = false;
-        });
+      this.matchService.submitRevision(this.existingQueryService.currentQuery.id)
+      .then(() => {
+        this.existingQueryService.updateQueryStateToProcessFinalized()
```

Comment:
By not returning ` this.existingQueryService.updateQueryStateToProcessFinalized`, you are braking the promise chain started by `this.matchService.submitRevision()`.  Everything wrapped up in this new promise chain will not have a guarantee of executing after `this.matchService.submitRevision()`.  Further, now that the inner chain is not longer a part of the outer chain, if the outer chain fails, `catch()` will never be entered.